### ISSUE TITLE
generator causing false positive using-constant-test

### ISSUE THREAD

**Issue by jwilk** [PyCQA/pylint#10080]
### Bug description

```python
def f():
    yield input()

[x] = f()
if x:
    pass
```

### Command used

```shell
pylint --disable C const.py 
```

### Pylint output

```
const.py:5:3: W0125: Using a conditional statement with a constant value (using-constant-test)
```

### Expected behavior

no warnings reported

### Pylint version

```
pylint 3.3.1
astroid 3.3.5
Python 3.13.0 (main, Oct  8 2024, 08:51:27) [GCC 9.4.0]
```
